src/pages/Transfer/CrossChainTransferForm/CrossChainTransferForm.tsx

```diff
@@ -35,125 +43,255 @@ import {
 
-  const { getBalance } = useGetBalances();
   const prices = useGetPrices();
   const { t } = useTranslation();
-  const { data: XCMBridge } = useXCMBridge();
+  const { getCurrencyFromTicker } = useGetCurrencies(true);
 
+  const accountId = useAccountId();
   const { accounts } = useSubstrateSecureState();
 
-  useEffect(() => {
-    if (!XCMBridge) return;
+  const { data, getDestinationChains, originatingChains, getAvailableTokens } = useXCMBridge();
+
+  const schema: CrossChainTransferValidationParams = {
+    [CROSS_CHAIN_TRANSFER_AMOUNT_FIELD]: {
+      minAmount: currentToken
+        ? newMonetaryAmount(currentToken.minTransferAmount, getCurrencyFromTicker(currentToken.value), true)
+        : undefined,
+      maxAmount: currentToken
+        ? newMonetaryAmount(currentToken.balance, getCurrencyFromTicker(currentToken.value), true)
+        : undefined
+    }
+  };
 
-    const availableChains = XCMBridge.bridge.adapters.map((adapter: any) => {
-      return {
-        display: adapter.chain.display,
-        id: adapter.chain.id
-      };
-    });
+  const mutateXcmTransfer = async (formData: CrossChainTransferFormData) => {
+    if (!data || !formData || !currentToken) return;
 
-    setTestChains(availableChains);
-  }, [XCMBridge]);
+    const { signer } = await web3FromAddress(formData[CROSS_CHAIN_TRANSFER_TO_ACCOUNT_FIELD] as string);
+    const adapter = data.bridge.findAdapter(formData[CROSS_CHAIN_TRANSFER_FROM_FIELD] as ChainName);
+    const apiPromise = data.provider.getApiPromise(formData[CROSS_CHAIN_TRANSFER_FROM_FIELD] as string);
 
-  const handleSubmit = () => {
-    console.log('submit');
-  };
+    apiPromise.setSigner(signer);
+    adapter.setApi(apiPromise);
 
-  const governanceBalance = getBalance(GOVERNANCE_TOKEN.ticker)?.free || newMonetaryAmount(0, GOVERNANCE_TOKEN);
-  const balance = getBalance(GOVERNANCE_TOKEN.ticker)?.transferable;
+    const transferAmount = newMonetaryAmount(
+      form.values[CROSS_CHAIN_TRANSFER_AMOUNT_FIELD] || 0,
+      getCurrencyFromTicker(currentToken.value),
+      true
+    );
 
-  const schema: CrossChainTransferValidationParams = {
-    [CROSS_CHAIN_TRANSFER_AMOUNT_FIELD]: {
-      governanceBalance,
-      minAmount: newMonetaryAmount(1, GOVERNANCE_TOKEN),
-      maxAmount: balance || newMonetaryAmount(0, GOVERNANCE_TOKEN),
-      transactionFee: TRANSACTION_FEE_AMOUNT
-    }
+    const transferAmountString = transferAmount.toString(true);
+    const transferAmountDecimals = transferAmount.currency.decimals;
+
+    const tx = adapter.createTx({
+      amount: FixedPointNumber.fromInner(transferAmountString, transferAmountDecimals),
+      to: formData[CROSS_CHAIN_TRANSFER_TO_FIELD],
+      token: formData[CROSS_CHAIN_TRANSFER_TOKEN_FIELD],
+      address: formData[CROSS_CHAIN_TRANSFER_TO_ACCOUNT_FIELD]
+    } as CrossChainTransferParams);
+
+    const inBlockStatus = getExtrinsicStatus('InBlock');
+
+    await DefaultTransactionAPI.sendLogged(
```

Comment:
I think this is duplicate logic of `submitExtrinsic`

Reply:
@peterslany could you comment on this? This is the same implementation as the current XCM version, which predates the `submitExtrinsic` work. Can this transaction implementation be simplified, do you think?

Reply:
@tomjeatt Yes I think this can be simplified by passing the `tx` into `submitExtrinsic` (as long as the account that is supposed to sign the extrinsic: `formData[CROSS_CHAIN_TRANSFER_TO_ACCOUNT_FIELD]` is the same as account connected to dapp, which is true in this case I think)

```typescript
await submitExtrinsic({extrinsic: tx});
```